tools/fastdds/discovery/parser.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Entry point of discovery sub-command."""
+
+import os
+import re
+import subprocess
+import sys
+from pathlib import Path
+
+
+class Parser:
+    """Discovery server tool parser."""
+
+    __help_message = 'fastdds discovery <discovery-args>\n\n'
+
+    def __find_tool_path(self):
+        """Calculate the path to the fast-discovery-server tool.
+
+        Returns
+        str : Full path to the executable
```

Comment:
In general, I think you have a mix of conventions for documenting inputs and outputs. I'd stick to the reST personally